### PR TITLE
[NM-912] base uri validation

### PR DESCRIPTION
**Pull request by resaairdropped**
`baseUri` validation

### CODE REVIEW

**Review by KevinH2810 — CHANGES_REQUESTED** [review]
please add unit test to prove that the function is working as expected

**Review by KevinH2810 — APPROVED** [review]

File: __tests__/utils/string.test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+describe('utils/string', () => {
```
Comment (rashgaroth) [inline __tests__/utils/string.test.ts:1]: please add the another case like:
- https://example.com
- http://example.com
- http://127.0.0.1:3002
- https://example.com/
- https://www.example.com

File: src/apis/tenant/index.ts [hunk 2]
```diff
@@ -264,7 +269,7 @@ export class TenantApis extends BaseApi {
       });
       return data;
     } catch (error: any) {
-      console.log(error, "@error?");
+      console.log(error, '@error?');
```
Comment (rashgaroth) [inline src/apis/tenant/index.ts]: could you please remove this also

**Review by rashgaroth — CHANGES_REQUESTED** [review]

File: src/apis/tenant/index.ts
Comment (capt4ce) [inline src/apis/tenant/index.ts:1]: can you check your formatter? It seems the changes in this file is just about formatting

File: src/utils/string.ts
Comment (capt4ce) [inline src/utils/string.ts:1]: as I remember we also need a function to call a request to the uri to check if the metadata format is correct, right @rashgaroth ?

**Review by capt4ce — CHANGES_REQUESTED** [review]

File: __tests__/utils/string.test.ts (hunk 1 above)
Comment (resaairdropped) [reply __tests__/utils/string.test.ts:1]: fix on last commit

File: src/apis/tenant/index.ts (hunk 2 above)
Comment (resaairdropped) [reply src/apis/tenant/index.ts]: fix on last commit

File: src/utils/string.ts
Comment (resaairdropped) [reply src/utils/string.ts:1]: will do on different ticket

**Review by rashgaroth — APPROVED** [review]

**Pull request merged**